core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -123,6 +123,16 @@ public static Object fromReader(Reader reader) throws IOException {
             throw new JsonParseException("document doesn't start with a valid json element : "
                     + initialToken, jp.getCurrentLocation());
         }
+        
+        JsonToken t ;
+        try { t = jp.nextToken(); }
```

Comment:
Yes, I think embedded JSON should be extracted before being parsed. In the case of HTML this is done by extracting the entire contents of a script tag, so the contents should be a single full JSON document at that point.

For now, checking for the three possible cases after parsing should work for giving an error instead of silently succeeding. Jackson is fairly generic (I am using the Jackson CSV parser in other projects), so it may be allowing cases like this for other formats, and not failing specifically in the JSON case.